pwiz_tools/Skyline/SkylineFiles.cs

```diff
@@ -894,6 +895,146 @@ private bool CheckResults(SrmDocument document, string path, FormEx parent)
+                }
+                
+            }
+        }
+
+        public void OpenFromPanorama()
+        {
+            var servers = Settings.Default.ServerList;
+            if (servers.Count == 0)
+            {
+                DialogResult buttonPress = MultiButtonMsgDlg.Show(
+                    this,
+                    TextUtil.LineSeparate(
+                        Resources.SkylineWindow_ShowPublishDlg_There_are_no_Panorama_servers_to_upload_to,
+                        Resources.SkylineWindow_ShowPublishDlg_Press_Register_to_register_for_a_project_on_PanoramaWeb_,
+                        Resources.SkylineWindow_ShowPublishDlg_Press_Continue_to_use_the_server_of_your_choice_),
+                    Resources.SkylineWindow_ShowPublishDlg_Register, Resources.SkylineWindow_ShowPublishDlg_Continue,
+                    true);
+                if (buttonPress == DialogResult.Cancel)
+                    return;
+
+                object tag = null;
+                if (buttonPress == DialogResult.Yes)
+                {
+                    // person intends to register                   
+                    WebHelpers.OpenLink(this, @"https://panoramaweb.org/signup.url");
+                    tag = true;
+                }
+
+                var serverPanoramaWeb = new Server(PanoramaUtil.PANORAMA_WEB, string.Empty, string.Empty);
+                var newServer = servers.EditItem(this, serverPanoramaWeb, null, tag);
+                if (newServer == null)
+                    return;
+
+                servers.Add(newServer);
+            }
+
+            var panoramaServers = servers.Cast<PanoramaServer>().ToList();
+
+            var state = string.Empty;
+            if (!string.IsNullOrEmpty(Settings.Default.FileExpansion))
+            {
+                state = Settings.Default.FileExpansion;
+            }
+
+            try
+            {
+                using var dlg = new FilePicker(panoramaServers, false, state, true);
+                using (var longWaitDlg = new LongWaitDlg
+                       {
+                           Text = "Loading remote server folders",
+                       })
+                {
+                    longWaitDlg.PerformWork(this, 0,
+                        () => dlg.InitializeDialog());
+                    if (longWaitDlg.IsCanceled)
+                        return;
+                }
+                if (dlg.ShowDialog() != DialogResult.Cancel)
+                {
+                    var folderPath = string.Empty;
+                    if (!string.IsNullOrEmpty(Settings.Default.LastFolderPath))
+                    {
+                        folderPath = Settings.Default.LastFolderPath;
+                    }
+                    var curServer = dlg.ActiveServer;
+                    var panoramaClient = new WebPanoramaClient(curServer.URI);
+
+                    var downloadPath = panoramaClient.SaveFile(dlg.FileName, folderPath);

```

Comment:
Since SaveFile is not interacting with the Panorama server perhaps it should not be a method in WebPanoramaClient. You could try replacing it here with:
```
string downloadPath;
using (var saveAsDlg = new SaveFileDialog
                                     {
                                          FileName = dlg.FileName,
                                          DefaultExt = SrmDocumentSharing.EXT_SKY_ZIP,
                                          Filter = TextUtil.FileDialogFilter(Resources.SrmDocument_FILTER_DOC_Skyline_Documents, SrmDocumentSharing.EXT_SKY_ZIP),
                                          InitialDirectory = folderPath,
                                          OverwritePrompt = true,
                                      })
 {
     if (saveAsDlg.ShowDialog(this) != DialogResult.OK)
      {
          return;
      }
      downloadPath = saveAsDlg.FileName;
 }
```

Reply:
I talked to Brendan about SaveFileDialog vs FolderBrowserDialog last week and he said for now FolderBrowserDialog is fine, but I went ahead and took the SaveFile method out of WebPanoramaClient and added the code you suggested into SkylineFiles so now it's using the SaveFileDialog. Hopefully that is fine but it would be an easy change to go back to FolderBrowserDialog if Brendan prefers that!